src/Geometry/Primitives/NonCSG/PlanarShapes.jl

```diff
@@ -0,0 +1,22 @@
+# MIT license
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE in the project root for full license information.
+
+
+"""The PlanarShapes interface:
+
+`distancefromplane(p::PlanarShapes,point)`  returns distance of the point from the plane the planar shape lies within
+`normal(p::PlanarShapes)` returns normal of plane
+`interface(p::PlanarShapes)` returns optical interface of plane
+`vertices(p::PlanarShapes)` returns vertices of shape. For Ellipse this is an approximation
+"""
+abstract type PlanarShapes{T} <: Surface{T} end
+
+"""All planar shapes lie on a plane. This function computes the distance from a point to that plane. This is a signed distance. If the point is on the positive side of the plane (the side the normal points toward) the distance will be positive, otherwise negative or 0 if the point lies in the plane."""
+distancefromplane(p::PlanarShapes, point::SVector{3}) = distancefromplane(p.plane,point)
```

Comment:
Think of this as the default function that you use if your PlanarShape object has a plane,normal,interface field. If it doesn't or you need to do some computation to generate the plane,normal,interface information then you create a more specialized function to override this one:

plane(a::Rectangle) = something different than a.plane

Otherwise we have must write 3 functions for 6 different PlanarShape types all of which look exactly the same:

plane(a::Rectangle) = a.plane
plane(a::Ellipse) = a.plane
...

This seems redundant and error prone.

Would it be good enough if I comment these functions to indicate that they are intended as defaults assuming you define these standard fields?

Reply:
@galran what did you think of my suggestion in the previous comment?